src/vs/workbench/contrib/terminalContrib/suggest/browser/terminalCompletionService.ts

```diff
@@ -204,12 +205,7 @@ export class TerminalCompletionService extends Disposable implements ITerminalCo
 
 		const resourceCompletions: ITerminalCompletion[] = [];
 		const endsWithSpace = promptValue.substring(0, cursorPosition).endsWith(' ');
-		let lastWord;
-		if (endsWithSpace) {
-			lastWord = '';
-		} else {
-			lastWord = promptValue.substring(0, cursorPosition).trim().split(' ').at(-1) ?? '';
-		}
+		const lastWord = endsWithSpace ? '' : promptValue.substring(0, cursorPosition + 1).trim().split(' ').at(-1) ?? '';
```

Comment:
What is the purpose of the `+ 1` here? Also the trim might cause bugs, I don't think it does anything atm due to the `endsWithSpace` check

Reply:
substring end value is not inclusive, thus the plus one

Reply:
But cursorPosition from promptInputState is the position of the cursor, so substring should get the text from 0 to where the cursor is (after the last char). Shouldn't it be the same as `endsWithSpace`?